src/content/docs/fr/reference/integrations-reference.mdx

```diff
@@ -0,0 +1,727 @@
+}
+```
+
+#### Option `pages`
+
+**Type :** `{ pathname: string }[]`
+
+Une liste de toutes les pages générées. Il s'agit d'un objet avec une propriété.
+
+- `pathname` - le chemin finalisé de la page.
+
+## Autoriser l'installation avec `astro add`
+
+[La commande `astro add`](/fr/reference/cli-reference/#astro-add) permet aux utilisateurs d'ajouter facilement des intégrations et des adaptateurs à leur projet. Si vous voulez que _votre_ intégration soit installable avec cet outil, **ajoutez `astro-integration` au champ `keywords` dans votre `package.json`** :
+
+```json
+{
+  "name": "example",
+  "keywords": ["astro-integration"],
+}
+```
+
+Une fois que vous avez [publié votre intégration sur npm](https://docs.npmjs.com/cli/v8/commands/npm-publish), lancer `astro add example` installera votre paquet avec toutes les dépendances spécifiées dans votre `package.json`. Cela appliquera également votre intégration au `astro.config` de l'utilisateur comme suit :
+
+```diff
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
++ import example from 'example';
+
+export default defineConfig({
++  integrations: [example()],
+})
+```
+
+:::caution
+Cela suppose que votre définition d'intégration est 1) une exportation `default` et 2) une fonction. Assurez-vous que c'est le cas avant d'ajouter le mot-clé `astro-integration` !
+:::
+
+## `AstroIntegrationLogger`
+
+Une instance du logger Astro, utile pour écrire des logs. Ce logger utilise le même [niveau de log](/fr/reference/cli-reference/#--verbose)
+configuré via le CLI.
+
+**Méthodes disponibles** pour écrire dans le terminal :
+- `logger.info("Message")`;
+- `logger.warn("Message")`;
+- `logger.error("Message")`;
+- `logger.debug("Message")`;
+
+**Méthodes disponibles** pour écrire dans le terminal :
+
+```ts title="integration.ts" {8}
+import type { AstroIntegration } from "astro";
+export function formatIntegration(): AstroIntegration {
+    return {
+        name: "astro-format",
+        hooks: {
+            "astro:build:done": ({ logger }) => {
+                // fait quelque chose
+                logger.info("Integration ready.");
+            }
+        }
+    }
+}
+```
+
+L'exemple ci-dessus enregistre un message qui comprend le message `info` fourni :
+
+```shell
+[astro-format] Integration ready.
+```
+
+Pour enregistrer certains messages avec un libellé différent, utilisez la méthode `.fork` pour spécifier une alternative au `name` par défaut :
+
+```ts title="integration.ts" ".fork"
+import type { AstroIntegration } from "astro";
+export function formatIntegration(): AstroIntegration {
+    return {
+        name: "astro-format",
+        hooks: {
+            "astro:config:done": ({ logger }) => {
+                // do something
+                logger.info("Integration ready.");
+            },
+            "astro:build:done": ({ logger }) => {
+                const buildLogger = logger.fork("astro-format/build");
+                // do something
```

Comment:
The previous occurence is translated with `// fait quelque chose` so I think it should also be translated.